### PR TITLE
Change master to advertise external IP in kubernetes service.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Change master to advertise external IP in kubernetes service.
In effect, in HA mode in case of multiple masters, IP of external load
balancer will be advertise in kubernetes service.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36346)

### CODE REVIEW

**Comment by jszczepkowski** [comment]
This PR is needed for HA master. It replaces #36057: instead of trying to keep up-to-date list of kubernetes apiserver in kubernetes service, we will direct all traffic to the external ip:
- in one master cluster the ip points to the single master,
- in multi-master cluster the ip points to the load balancer in-front of the masters.

This PR make kubernetes service consistent with how kubelets communicate with the masters (by public external IP, #33655). It also make consistent kubernetes service in GCE with kubernetes service in GKE (points to public external IP).

File: cluster/common.sh [hunk 1]
```diff
@@ -701,6 +701,7 @@ MANIFEST_URL_HEADER: $(yaml-quote ${MANIFEST_URL_HEADER:-})
 NUM_NODES: $(yaml-quote ${NUM_NODES})
 STORAGE_BACKEND: $(yaml-quote ${STORAGE_BACKEND:-etcd2})
 ENABLE_GARBAGE_COLLECTOR: $(yaml-quote ${ENABLE_GARBAGE_COLLECTOR:-})
+MASTER_RESERVED_IP: $(yaml-quote ${MASTER_RESERVED_IP:-})
```
Comment (fgrzadkowski) [inline cluster/common.sh]: As discussed offline I think we should not use env var `MASTER_RESERVED_IP` as it's GCE specific. We should use something that will make sense also on other cloud providers.

Comment (jszczepkowski) [inline cluster/common.sh]: Fixed: introduced new var MASTER_ADVERTISE_ADDRESS

**Comment by fgrzadkowski** [comment]
It LGTM, but please wait for @roberthbailey to review it as well.

**Comment by roberthbailey** [comment]
lgtm

**Comment by saad-ali** [comment]
Applying do-not-merge until the post-code freeze merge exception for this feature has been approved.

https://groups.google.com/forum/#!topic/kubernetes-milestone-burndown/DTZu98YNuUE

**Comment by saad-ali** [comment]
> Applying do-not-merge until the post-code freeze merge exception for this feature for 1.5 has been approved.
>
> https://groups.google.com/forum/#!topic/kubernetes-milestone-burndown/DTZu98YNuUE

The request for exception to 1.5 release code freeze for the "Simplify HA setup for master on GCE" feature has been approved. Adding 1.5 milestone and removing `do-not-merge` label.

**Comment by jszczepkowski** [comment]
Part of https://github.com/kubernetes/features/issues/48

**Pull request merged**